File: apps/api/src/services/sms.service.ts
import axios from "axios";

export async function sendOTPSms(phone: string, otp: string): Promise<boolean> {
  try {
    const apiKey = process.env.SMS_API_KEY;
    if (!apiKey) {
      console.log(`[DEV MODE] OTP for ${phone}: ${otp}`);
      return true;
    }
    const url = `https://2factor.in/API/V1/${apiKey}/SMS/${phone}/${otp}`;
    const { data } = await axios.get(url);
    if (data.Status === "Success") {
      console.log(`OTP sent to ${phone}`);
      return true;
    }
    console.error("SMS failed:", data);
    return false;
  } catch (error) {
    console.error("SMS error, falling back to console:", error);
    console.log(`[DEV MODE] OTP for ${phone}: ${otp}`);
    return false;
  }
}

File: backend/app/services/sms_service.py
import os

import requests

from .. import config


async def send_otp_sms(phone: str, otp: str) -> bool:
    api_key = config.SMS_API_KEY
    if not api_key:
        print(f"[DEV MODE] OTP for {phone}: {otp}")
        return True
    url = f"https://2factor.in/API/V1/{api_key}/SMS/{phone}/{otp}"
    try:
        resp = requests.get(url, timeout=15)
        data = resp.json()
        if data.get("Status") == "Success":
            print(f"OTP sent to {phone}")
            return True
        print("SMS failed:", data)
        return False
    except Exception:
        print(f"[DEV MODE] OTP for {phone}: {otp}")
        return False
